app/javascript/src/components/Dashboard/Notes/NoteList.jsx

```diff
@@ -17,15 +18,39 @@ export const COLUMN_DATA = [
   },
 ];
 
-export default function NoteList({ notes = [] }) {
+export default function NoteList({
+  setSelectedNoteId,
+  selectedNoteId,
+  setShowDeleteAlert,
+  notes = [],
+  fetchNotes,
+}) {
+  const [showEditNote, setShowEditNote] = useState(false);
+  const [selectedNote, setSelectedNote] = useState(
+    notes.find(n => n.id === selectedNoteId)
```

Comment:
`notes.find(note => note.id === selectedNoteId)`
Also, it's always better to extract such logic to a function.